### PR TITLE
Added SA impersonation for BigQuery client

### PR DESCRIPTION
**Pull request by shnapz**
Implementing [#4591](https://github.com/spotify/scio/issues/4591)
In order to configure impersonation two new SBT options were added:

- bigquery.act_as (target SA principal)
- bigquery.act_as_lifetime (a duration in seconds of a target SA temporary credentials lifetime)

### CODE REVIEW

File: scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/client/BigQueryConfig.scala [hunk 1]
```diff
@@ -70,6 +72,14 @@ object BigQueryConfig {
   def readTimeoutMs: Option[Int] =
     BigQuerySysProps.ReadTimeoutMs.valueOption.map(_.toInt)
 
+  def impersonationInfo: Option[ImpersonationInfo] = {
+    BigQuerySysProps.ActAs.valueOption.map { actAs =>
+      val lifetime =
+        BigQuerySysProps.ImpersonationLifetimeSec.valueOption.flatMap(x => Try(x.toInt).toOption)
+      ImpersonationInfo(actAs, lifetime.getOrElse(3600))
```
Comment (clairemcginty) [inline scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/client/BigQueryConfig.scala:79]:
```suggestion
      ImpersonationInfo(actAs, lifetime.getOrElse(0))
```

I would set the default here to 0, that seems to be the Google library's hint to use its default value: https://github.com/googleapis/google-auth-library-java/blob/main/oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java#L679

**Review by clairemcginty — APPROVED** [review]
one small suggestion but otherwise, looks great! 🎉

**Review by clairemcginty — APPROVED** [review]

File: scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/client/BigQueryConfig.scala [hunk 2]
```diff
@@ -70,6 +72,14 @@ object BigQueryConfig {
   def readTimeoutMs: Option[Int] =
     BigQuerySysProps.ReadTimeoutMs.valueOption.map(_.toInt)
 
+  def impersonationInfo: Option[ImpersonationInfo] = {
+    BigQuerySysProps.ActAs.valueOption.map { actAs =>
+      val lifetime =
+        BigQuerySysProps.ImpersonationLifetimeSec.valueOption.flatMap(x => Try(x.toInt).toOption)
```
Comment (RustedBones) [inline scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/client/BigQueryConfig.scala:78]: do we want to silently default the `lifetime` if invalid ? can't we let it crash ?

**Review by RustedBones — COMMENTED** [review]
Nice work. And as always good effort on the doc :+1:

File: scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/client/BigQueryConfig.scala (hunk 2 above)
Comment (shnapz) [reply scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/client/BigQueryConfig.scala:78]: I think you are right

**Comment by shnapz** [comment]
@clairemcginty @RustedBones made all config reads parsing fail fast

**Pull request merged**